File: python/solutions/p005_smallest_mult.py
#!/usr/bin/env python3 -OO

"""Smallest multiple
Problem 5
2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.

What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?
"""

def run(num, div):
    guess = 0
    found = False
    while not found:
        found = True
        guess += num
        for idx in range(1, num + 1):
            quotient = guess / idx
            remainder = guess % idx
            if remainder != 0 or quotient % div != 0:
                found = False

    return guess

File: rust/src/p005_smallest_mult.rs
//! Smallest multiple
//! Problem 5
//! 2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
//!
//! What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?


fn run(val: i32) -> i32 {
    let mut guess = 0;
    let mut found = false;
    while !found {
        found = true;
        guess += val;
        for num in 1..=val {
           if guess % num != 0 {
               found = false;
               break;
           }
        }
    }
    guess
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_p005() {
        assert_eq!(run(10), 2520);
    }
}
